c7n/resources/cw.py

```diff
@@ -35,6 +37,71 @@ class resource_type(object):
         dimension = None
 
 
+@Alarm.filter_registry.register('age')
+class AlarmAge(AgeFilter):
+    """CloudWatch Alarm Age Filter
+
+    Filters a CloudWatch Alarm based on the age of the last alarm state
+    change (in days)
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: unchanged-alarm-month-old
+                resource: ebs-snapshot
+                filters:
+                  - type: age
+                    days: 30
+                    op: ge
+    """
+
+    schema = type_schema(
+        'age',
+        days={'type': 'number'},
+        op={'type': 'string', 'enum': OPERATORS.keys()})
+    date_attribute = 'StateUpdatedTimestamp'
+
+
+@Alarm.action_registry.register('delete')
+class AlarmDelete(BaseAction):
+    """
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: cloudwatch-delete-stale-alarms
+                resource: alarm
+                filters:
+                  - type: age
+                    days: 30
+                    op: ge
+                  - StateValue: INSUFFICIENT_DATA
+                actions:
+                  - delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('cloudwatch:DeleteAlarms',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('cloudwatch')
+        alarms = []
+
+        for i, r in enumerate(resources, start=1):
```

Comment:
for resource_set in chunks(resources, 100): client.delete_alarms(AlarmNames=[r['Name'] for r in resource_set])